lonestar/analytics/cpu/triangles/Triangles.cpp

```diff
@@ -354,46 +354,98 @@ void edgeIteratingAlgo(Graph& graph) {
   std::cout << "NumTriangles: " << numTriangles.reduce() << "\n";
 }
 
-void makeGraph(Graph& graph, const std::string& triangleFilename) {
-  typedef galois::graphs::FileGraph G;
-  typedef G::GraphNode N;
-
-  G initial, permuted;
-
+//! Sorts read graph by degree (high degree nodes are reindexed to beginning)
+void makeSortedGraph(Graph& graph) {
+  // read original graph
+  galois::graphs::FileGraph initial;
   initial.fromFileInterleaved<void>(inputFilename);
 
-  // Getting around lack of resize for deque
-  std::deque<N> nodes;
-  std::copy(initial.begin(), initial.end(), std::back_inserter(nodes));
-
-  /* Sort by degree:
-   *  DegreeLess: Sorts in the ascending order of node degrees
-   *  DegreeGreater: Sorts in the descending order of the node degrees
-   *
-   *  The order of sorting has a huge impact on performance
-   *  For this algorithm, sorting in descending order delivers the
-   *  best performance due to the way ties are broken.
-   */
-  galois::ParallelSTL::sort(nodes.begin(), nodes.end(),
```

Comment:
New sort has the same signature as this, right?

Reply:
yes